src/sentry/models/release.py

```diff
@@ -283,43 +279,15 @@ def set_refs(self, refs, user, fetch_commits=False):
                     'commit': commit,
                 }
             )
-            if fetch_commits:
-                try:
-                    provider_cls = bindings.get('repository.provider').get(repo.provider)
-                except KeyError:
-                    continue
-
-                # if previous commit isn't provided, try to get from
-                # previous release otherwise, give up
-                if ref.get('previousCommit'):
-                    start_sha = ref['previousCommit']
-                elif prev_release:
-                    try:
-                        start_sha = Commit.objects.filter(
-                            organization_id=self.organization_id,
-                            releaseheadcommit__release=prev_release,
-                            repository_id=repo.id,
-                        ).values_list('key', flat=True)[0]
-                    except IndexError:
-                        continue
-                else:
-                    continue
-
-                end_sha = commit.key
-                provider = provider_cls(id=repo.provider)
-                try:
-                    repo_commits = provider.compare_commits(
-                        repo, start_sha, end_sha, actor=user
-                    )
-                except NotImplementedError:
-                    pass
-                except (PluginError, InvalidIdentity) as e:
-                    logger.exception(six.text_type(e))
-                else:
-                    commit_list.extend(repo_commits)
-
-            if commit_list:
-                self.set_commits(commit_list)
+        if fetch:
+            fetch_commits.apply_async(
+                kwargs={
+                    'release_id': self.id,
+                    'user_id': user.id,
+                    'refs': refs,
```

Comment:
How large are these `ref` objects? Can we slim them down to just the bits we need? It seems we only use `commit`, `repository`, and `previousCommit`. Is there a lot of other junk piled on here?

If not, we're good. Otherwise, I'd want to slim this down before throwing into the queue.